src/editor.c

```diff
@@ -857,13 +863,15 @@ static void on_char_added(GeanyEditor *editor, SCNotification *nt)
 		case ':':	/* C/C++ class:: syntax */
 		/* tag autocompletion */
 		default:
-#if 0
-			if (! editor_start_auto_complete(editor, pos, FALSE))
-				request_reshowing_calltip(nt);
-#else
 			editor_start_auto_complete(editor, pos, FALSE);
-#endif
 	}
+
+	if (plugin_extension_calltips_provided(editor->document, NULL))
+		plugin_extension_calltips_show(editor->document, FALSE);
+
+	if (plugin_extension_autocomplete_provided(editor->document, NULL))
+		plugin_extension_autocomplete_perform(editor->document, FALSE);
```

Comment:
There's no need to manually call `_provided()`, at least not anymore, is there?
```suggestion
	plugin_extension_calltips_show(editor->document, FALSE);
	plugin_extension_autocomplete_perform(editor->document, FALSE);
```

Reply:
> There's no need to manually call _provided(), at least not anymore, is there?

No, there isn't, it's a left-over from the previous implementation.